clients/src/main/java/org/apache/kafka/clients/producer/MockProducer.java

```diff
@@ -208,7 +209,7 @@ public void abortTransaction() throws ProducerFencedException {
 
     private void verifyProducerState() {
         if (this.closed) {
-            throw new IllegalStateException("MockProducer is already closed.");
+            throw new KafkaException("MockProducer is already closed.");
```

Comment:
I think it's fine to leave this unchanged since it is only invoked at the start of the mock producer apis.

Reply:
Probably not a big deal, but I do see this being called from `MockProducer#send` for example so it might be worth keeping things consistent by throwing `KafkaException` as we do when `KafkaProducer#send` is called after close.